### PR TITLE
Remove bindings/usage of ttdnn creation ops 

### PR DESCRIPTION
**Pull request by umadevimcw**
### Ticket
#10644 

### Problem description
Remove and replace following list of ttdnn ops to ttnn
- arange
- empty
- full_like
- empty_like
- full
- zeros
- zeros_like
- ones
- ones_like
Remove unused header includes of composite fils

### What's changed
Replace `tt_lib.tensor.<op_name>` to `ttnn.<op_name>` in test and source file

### Checklist  

Yet to complete and test 

- [ ]  [All post-commit tests] https://github.com/tenstorrent/tt-metal/actions/runs/10207323967
- [ ]  [Nightly fast dispatch tests] https://github.com/tenstorrent/tt-metal/actions/runs/10187305833
- [ ] [post-commit] models tests https://github.com/tenstorrent/tt-metal/actions/runs/10187309355
- [ ] post-commit c++ https://github.com/tenstorrent/tt-metal/actions/runs/10187358589
- [ ] t3k unit https://github.com/tenstorrent/tt-metal/actions/runs/10187300327
- [ ] t3k demo https://github.com/tenstorrent/tt-metal/actions/runs/10187303788
- [ ] t3k frequent https://github.com/tenstorrent/tt-metal/actions/runs/10187302058
- [ ] model perf https://github.com/tenstorrent/tt-metal/actions/runs/10187388220
- [ ] device perf

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/test_creation.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def test_arange(device, start, end, step):
     input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.TILE_LAYOUT)
     input_tensor = ttnn.to_device(input_tensor, device)
 
-    output_tensor = ttnn.arange(input_tensor.shape[0], input_tensor.shape[1], input_tensor.shape[2], device)
+    output_tensor = ttnn.arange(input_tensor.shape[0], input_tensor.shape[1], input_tensor.shape[2])
```
Comment (ayerofieiev-tt) [inline tests/ttnn/unit_tests/operations/test_creation.py:160]: Device redundant here?

File: ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/backward/backward_ops.cpp [hunk 2]
```diff
@@ -0,0 +1,321 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttnn/deprecated/tt_dnn/op_library/backward/backward_ops.hpp"
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/backward/backward_ops.cpp:5]: I am not clear why is this here?

File: ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/backward/backward_ops.cpp~#10644: Replace ttlib full with ttnn full [hunk 3]
```diff
@@ -0,0 +1,321 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/backward/backward_ops.cpp~#10644: Replace ttlib full with ttnn full:3]: Remove?

File: ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/composite/composite_ops.cpp [hunk 4]
```diff
@@ -198,7 +198,7 @@ Tensor mac(const Tensor& input_a, float b, float c, const MemoryConfig& output_m
 
 Tensor _logical_andi(const Tensor& input_a, float immediate, const MemoryConfig& output_mem_config) {
     if (std::fpclassify(immediate) == FP_ZERO) {
-        return full_like(input_a, immediate, output_mem_config);
+        return ttnn::operations::creation::full_like(input_a, immediate);
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/composite/composite_ops.cpp:201]: this should be just `ttnn::full_like`, same in other places

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp [hunk 5]
```diff
@@ -271,21 +271,13 @@ std::vector<std::optional<Tensor>> _eq_bw(
     std::vector<std::optional<Tensor>> result;
 
     if (are_required_outputs.at(0)) {
-        if(input_grad.has_value()){
-            tt::tt_metal::zeros_like(cq_id, input, output_mem_config, input_grad);
-        } else {
-            input_grad = tt::tt_metal::zeros_like(cq_id, input, output_mem_config);
-        }
+        input_grad = ttnn::operations::creation::zeros_like(input, input.get_dtype(), input.get_layout(), std::nullopt, output_mem_config);
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:274]: Can you clarify this change please? I am not sure why

Comment (ayerofieiev-tt) [reply ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:274]: Is it because `    if (are_required_outputs.at(0)) {` this check is enough?

File: ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp [hunk 6]
```diff
@@ -1431,24 +1428,26 @@ std::vector<Tensor> _repeat_bw(
     } else if (shape[0] > 1) {
         std::vector<int64_t> dim = {0};
         TT_FATAL(shape[1] == 1 && shape[2] == 1 && shape[3] == 1 && "repeat[1], [2], [3] should be 1");
-        const tt::tt_metal::Shape required = {1, shape_wh[1], shape_wh[2], shape_wh[3]};
+        std::array<std::uint32_t, 4> intended_shape_array = {1, shape_wh[1], shape_wh[2], shape_wh[3]};
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp:1431]: I think we have a type `Array4D` available, fyi

File: ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp [hunk 7]
```diff
@@ -1422,6 +1418,7 @@ std::vector<Tensor> _repeat_bw(
 
     auto shape_wh = input.get_legacy_shape();
     TT_FATAL(shape_wh[0] == 1 && "input shape[0] should be 1");
+    auto ttnn_device = input.device();
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp:1421]: can't see where this variable is used

File: ttnn/cpp/ttnn/deprecated/tt_lib/csrc/tt_lib_bindings_tensor_composite_ops.cpp [hunk 8]
```diff
@@ -235,83 +235,6 @@ void TensorModuleCompositeOPs(py::module& m_tensor) {
                 "output_mem_config", "Layout of tensor in TT Accelerator device memory banks", "MemoryConfig", "Default is interleaved in DRAM", "No"
         )doc");
 
-    m_tensor.def(
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/deprecated/tt_lib/csrc/tt_lib_bindings_tensor_composite_ops.cpp:238]: There are bindings already available in ttnn?

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp (hunk 5 above)
Comment (KalaivaniMCW) [reply ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:274]: It is because ttnn.zeros_like doesn't have queue_id support while tt_eager zeros_like had.

File: ttnn/cpp/ttnn/deprecated/tt_lib/csrc/tt_lib_bindings_tensor_composite_ops.cpp (hunk 8 above)
Comment (KalaivaniMCW) [reply ttnn/cpp/ttnn/deprecated/tt_lib/csrc/tt_lib_bindings_tensor_composite_ops.cpp:190]: yes

File: ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp (hunk 7 above)
Comment (KalaivaniMCW) [reply ttnn/cpp/ttnn/operations/eltwise/unary_backward/device/unary_backward_op.cpp:1421]: to pass device to zeros()

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]

**Review by SeanNijjar — APPROVED** [review]
